packages/ethereum/cardstack/indexer.js

```diff
@@ -142,17 +140,15 @@ class Updater {
       for (let model of schema) {
         await ops.save(model.type, model.id, model);
       }
-    } else {
-      await ops.beginReplaceAll();
+      await ops.finishReplaceAll();
```

Comment:
If you don't have valid meta, or if the schema has changed, you are reindexing everything -- schema and content both.

So `{begin|finish}ReplaceAll` needs to wrap the whole indexing operation, not just the schema part. Otherwise you can end up with stale ethereum records still in the search index.

Reply:
ah, ok. since ethereum contracts never change schema (because the blockchain is immuatable like that), i wonder if it even makes sense to have a `{begin|finish}ReplaceAll`? 

It probably makes sense to reindex everything if there is no meta/or schema, as there should always be both after the first indexing. Not having meta or schema probably means the index has been wiped clean of our ethereum data source.

Reply:
I think we still need replaceAll. In general there are a bunch of things that might cause us to want to redo indexing work (like adding a computed field to an ethereum-backed record). The only way to convey that to the plugin is to get rid of its meta and force it to start again. Under those conditions, it should get rid of any of its older stuff in the search index.

Reply:
cool, I fixed the placement of the begin/finish replace all in the latest commit.